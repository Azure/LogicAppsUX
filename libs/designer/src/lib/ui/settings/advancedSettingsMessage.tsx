import { useIntl } from 'react-intl';
import { Link, Text } from '@fluentui/react-components';
import { bundleIcon, LinkSquare12Regular, LinkSquare12Filled } from '@fluentui/react-icons';
const NavigateIcon = bundleIcon(LinkSquare12Regular, LinkSquare12Filled);

export const AdvancedSettingsMessage = ({ shouldShowMessage }: { shouldShowMessage: boolean }) => {
  const intl = useIntl();

  const IntlMessage = {
    message: intl.formatMessage({
      defaultMessage:
        'Add advanced parameters to further customize how the language model responds. These advanced options are optional and can help you fine-tune the experience.',
      description: 'Description for agent advanced parameters',
      id: 'uohgys',
    }),
    messageLink: intl.formatMessage({
      defaultMessage: 'Guideline for agent advanced parameters',
      description: 'Description link for agent advanced parameters',
      id: 'ZGGkV+',
    }),
  };

  return (
    shouldShowMessage && (
      <div style={{ paddingBottom: '5px' }}>
        <Text style={{ fontSize: 12 }}>{IntlMessage.message} </Text>
        <Link href="https://aka.ms/LogicApps/Agents/parameters" target="_blank" style={{ fontSize: 12, fontStyle: 'italic' }}>
          {IntlMessage.messageLink}
          <NavigateIcon style={{ position: 'relative', top: '2px', left: '2px' }} />
        </Link>
      </div>
    )
  );
};
