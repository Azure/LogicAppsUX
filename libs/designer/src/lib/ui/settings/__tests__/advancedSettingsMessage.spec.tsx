// biome-ignore lint/correctness/noUnusedImports: <explanation>
import React from 'react';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import { AdvancedSettingsMessage } from '../advancedSettingsMessage';
import { describe, expect, it } from 'vitest';

describe('AdvancedSettingsMessage Component', () => {
  it('should not render anything when shouldShowMessage is false', () => {
    const { container } = render(
      <IntlProvider locale="en">
        <AdvancedSettingsMessage shouldShowMessage={false} />
      </IntlProvider>
    );
    // Expecting null because the component should not render anything
    expect(container.firstChild).toBeNull();
  });

  it('should render correctly and match snapshot when shouldShowMessage is true', () => {
    const { container } = render(
      <IntlProvider locale="en">
        <AdvancedSettingsMessage shouldShowMessage={true} />
      </IntlProvider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
