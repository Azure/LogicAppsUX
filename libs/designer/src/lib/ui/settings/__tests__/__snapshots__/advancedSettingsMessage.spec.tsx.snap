// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AdvancedSettingsMessage Component > should render correctly and match snapshot when shouldShowMessage is true 1`] = `
<div
  style="padding-bottom: 5px;"
>
  <span
    class="fui-Text ___c56hd20_1xy0w11 fk6fouc fkhj508 f1i3iumi figsok6 fpgzoln f1w7gpdv f6juhto f1gl81tg f2jf649"
    style="font-size: 12px;"
  >
    Add advanced parameters to further customize how the language model responds. These advanced options are optional and can help you fine-tune the experience.
     
  </span>
  <a
    class="fui-Link ___1fq7916_1xye54t f2hkw1w f3rmtva f1ewtqcl fyind8e f1k6fduh f1w7gpdv fk6fouc fjoy568 figsok6 f1s184ao f1mk8lai fnbmjn9 f1o700av f1iuv45f f1cmlufx f9n3di6 f1ids18y f1tx3yz7 f1deo86v f1eh06m1 f1iescvh fhgqx19 f1olyrje f1p93eir f1nev41a"
    href="https://aka.ms/LogicApps/Agents/parameters"
    style="font-size: 12px; font-style: italic;"
    tabindex="0"
    target="_blank"
  >
    Guideline for agent advanced parameters
    <svg
      aria-hidden="true"
      class="fui-Icon-filled ___1vjqft9_1uxjcqz fjseox fez10in fg4l7m0"
      fill="currentColor"
      height="12"
      style="position: relative; top: 2px; left: 2px;"
      viewBox="0 0 12 12"
      width="12"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1 3c0-1.1.9-2 2-2h3a2 2 0 0 1 2 2v3a2 2 0 0 1-2 2V7a1 1 0 0 0 1-1V3a1 1 0 0 0-1-1H3a1 1 0 0 0-1 1v3a1 1 0 0 0 1 1v1a2 2 0 0 1-2-2V3Zm3 6c0 1.1.9 2 2 2h3a2 2 0 0 0 2-2V6a2 2 0 0 0-2-2v1a1 1 0 0 1 1 1v3a1 1 0 0 1-1 1H6a1 1 0 0 1-1-1V6a1 1 0 0 1 1-1V4a2 2 0 0 0-2 2v3Z"
        fill="currentColor"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="fui-Icon-regular ___12fm75w_1vnroxd f1w7gpdv fez10in fg4l7m0"
      fill="currentColor"
      height="12"
      style="position: relative; top: 2px; left: 2px;"
      viewBox="0 0 12 12"
      width="12"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1 3.25C1 2 2 1 3.25 1h2.5c1.22 0 2.2.96 2.25 2.16v2.68a2.25 2.25 0 0 1-1.5 2.03V3.25a.75.75 0 0 0-.75-.75h-2.5a.75.75 0 0 0-.75.75v2.5c0 .33.2.6.5.71v1.53a2.25 2.25 0 0 1-2-2.24v-2.5Zm3 5.5C4 9.99 5 11 6.25 11h2.5C9.99 11 11 10 11 8.75v-2.5c0-1.16-.88-2.11-2-2.24v1.53c.3.1.5.38.5.71v2.5c0 .41-.34.75-.75.75h-2.5a.75.75 0 0 1-.75-.75V4.13C4.63 4.43 4 5.27 4 6.25v2.5Z"
        fill="currentColor"
      />
    </svg>
  </a>
</div>
`;
